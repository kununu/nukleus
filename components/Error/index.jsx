import React, {PropTypes} from 'react';

import styles from './index.scss';

const Error = ({errorStyle, info, subInfo}) => (
  <span className={`${styles.error} ${styles[errorStyle]}`}>
    <strong className={styles.info}>{info}</strong>
    {subInfo}
  </span>
);

Error.propTypes = {
  errorStyle: PropTypes.oneOf(['inline']),
  info: PropTypes.string.isRequired,
  subInfo: PropTypes.string
};

Error.defaultProps = {
  errorStyle: 'inline',
  subInfo: ''
};

export default Error;
