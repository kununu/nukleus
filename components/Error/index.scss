@import '~kununu-theme/scss/base/variables';

.error {
  display: block;
  width: 100%;
  margin-top: $baseline-grid;
  padding: $padding-base-vertical $padding-base-horizontal;
  color: $white;
  background-color: $red;
  border-radius: $input-border-radius;
}

.info {
  display: block;
}
