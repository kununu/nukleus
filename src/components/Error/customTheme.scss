.errorContainer {
  box-sizing: border-box;
  display: block;
  width: 200px;
  padding: 10px;
  color: black;
  background-color: orange;
  border-radius: 10px;
  border: 2px solid darkorange;
}

.errorInfo {
  display: block;
}
