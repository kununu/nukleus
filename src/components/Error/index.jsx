import React from 'react';
import PropTypes from 'prop-types';

import styles from './index.scss';

const Error = ({info, subInfo}) => (
  <span className={styles.error}>
    <strong className={styles.info}>{info}</strong>
    {subInfo}
  </span>
);

Error.propTypes = {
  info: PropTypes.string.isRequired,
  subInfo: PropTypes.string,
};

Error.defaultProps = {
  subInfo: '',
};

export default Error;
