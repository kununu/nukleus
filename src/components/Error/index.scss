@import '~@kununu/kununu-theme-v2/scss/base/variables';

.errorContainer {
  display: block;
  width: 100%;
  margin-top: $baseline-grid;
  padding: $padding-base-vertical $padding-base-horizontal;
  color: $white;
  background-color: $red;
  border-radius: $input-border-radius;
}

.errorInfo {
  display: block;
}
