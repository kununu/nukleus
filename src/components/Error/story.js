import React from 'react';
import {storiesOf} from '@storybook/react';
import {defaultOptions} from 'defaultOptions';
import {withOptions} from '@storybook/addon-options';
import {withInfo} from '@storybook/addon-info';
import {withKnobs, text} from '@storybook/addon-knobs/react';

import ErrorComponent from './index';

storiesOf('Error', module)
  .addDecorator(withOptions({...defaultOptions}))
  .addDecorator(withKnobs)
  .addDecorator((story, context) => withInfo('Basic error block')(story)(context))
  .add('basics', () => (
    <ErrorComponent
      info={text('info', 'This is an error')}
      subInfo={text('subinfo', 'And I am some sub info')}
    />
  ));
