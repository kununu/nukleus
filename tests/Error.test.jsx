import React from 'react';
import renderer from 'react-test-renderer';
import Error from 'Error'; // eslint-disable-line import/no-unresolved

test('Renders an Error component without crashing', () => {
  const component = renderer.create(<Error info="Error" />);

  expect(component.toJSON()).toMatchSnapshot();
});

test('Renders an Error with a subInfo Text component without crashing', () => {
  const component = renderer.create(<Error
    info="Error"
    subInfo="Helpful hint"
  />);

  expect(component.toJSON()).toMatchSnapshot();
});

test('it can have an optional ID attribute', () => {
  const component = renderer.create(<Error
    info="Error"
    subInfo="Helpful hint"
    id="error-message"
  />);

  expect(component.toJSON()).toMatchSnapshot();
});
