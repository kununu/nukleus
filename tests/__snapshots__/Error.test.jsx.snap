// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders an Error component without crashing 1`] = `
<span
  className="errorContainer"
  id={null}
>
  <strong
    className="errorInfo"
  >
    Error
  </strong>
  
</span>
`;

exports[`Renders an Error with a subInfo Text component without crashing 1`] = `
<span
  className="errorContainer"
  id={null}
>
  <strong
    className="errorInfo"
  >
    Error
  </strong>
  Helpful hint
</span>
`;

exports[`it can have an optional ID attribute 1`] = `
<span
  className="errorContainer"
  id="error-message"
>
  <strong
    className="errorInfo"
  >
    Error
  </strong>
  Helpful hint
</span>
`;
